### PR TITLE
Pause resume idiom

### PR DESCRIPTION
**Pull request by kingsleyb**
## Summary
Idiom for manual_only trigger, demonstrating how to pause a task that can only be resumed with manual intervention. 

## Checklist

This PR:

- [ ] adds new tests (if appropriate)
- [ ] adds a change file in the `changes/` directory (if appropriate)
- [ ] updates docstrings for any new functions or function arguments, including `docs/outline.toml` for API reference docs (if appropriate)

### CODE REVIEW

File: docs/core/idioms/pause-for-approval.md [hunk 1]
```diff
@@ -0,0 +1,53 @@
+# Pause for Approval
```
Comment (joshmeek) [inline docs/core/idioms/pause-for-approval.md:1]: Could you add this to the list on `idioms.md`?

File: docs/core/idioms/pause-for-approval.md [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+
+A manual-only trigger that has been applied to a task will pause the task once it is reached. The task can only be resumed by an individual or event, either from the Prefect UI or GraphQL API.
```
Comment (cicdw) [inline docs/core/idioms/pause-for-approval.md:3]: I suggest adding a little more context prior to this line; something like:

There are many situations where a task requires some form of manual approval in order to run successfully.  Prefect has two built-in mechanisms for achieving this pattern: 
- a manual-only trigger that can be configured on a per task basis, and prevents that task from running unless a user approves it
- a `PAUSE` signal that can be raised programmatically when certain conditions are met

Both of these mechanisms mean that the task can only be resumed by an individual or event, either from the Prefect UI or GraphQL API.

File: docs/core/idioms/pause-for-approval.md [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+    return True
+
+@task
+def test(build_result):
+    print("test task")
+    build_result == True
+
+@task(trigger=manual_only)
+def deploy():
+    print("deploy task")
```
Comment (cicdw) [inline docs/core/idioms/pause-for-approval.md:28]: I suggest a docstring on this task that reinforces that it will only run if someone approves it

**Review by cicdw — CHANGES_REQUESTED** [review]

File: docs/core/idioms/pause-for-approval.md (hunk 1 above)
Comment (kingsleyb) [reply docs/core/idioms/pause-for-approval.md:1]: Added.

File: docs/core/idioms/pause-for-approval.md (hunk 2 above)
Comment (kingsleyb) [reply docs/core/idioms/pause-for-approval.md]: See changes.

**Review by cicdw — DISMISSED** [review]

File: docs/core/idioms/pause-for-approval.md (hunk 3 above)
Comment (kingsleyb) [reply docs/core/idioms/pause-for-approval.md:30]: added

**Review by cicdw — APPROVED** [review]

**Pull request merged**